fly_io/datadog_checks/fly_io/check.py

```diff
@@ -0,0 +1,233 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from six.moves.urllib.parse import quote_plus
+
+from datadog_checks.base import OpenMetricsBaseCheckV2
+
+from .config_models import ConfigMixin
+from .constants import (
+    APP_COUNT_METRIC,
+    MACHINE_COUNT_METRIC,
+    MACHINE_CPUS_METRIC,
+    MACHINE_GPUS_METRIC,
+    MACHINE_MEM_METRIC,
+    MACHINE_SWAP_SIZE_METRIC,
+    MACHINE_UP_STATE,
+    MACHINES_API_UP_METRIC,
+    VOLUME_BLOCK_SIZE_METRIC,
+    VOLUME_BLOCKS_AVAIL_METRIC,
+    VOLUME_BLOCKS_FREE_METRIC,
+    VOLUME_BLOCKS_METRIC,
+    VOLUME_CREATED_METRIC,
+    VOLUME_CREATED_STATE,
+    VOLUME_ENCRYPTED_METRIC,
+    VOLUME_SIZE_METRIC,
+)
+from .metrics import METRICS, RENAME_LABELS_MAP
+
+
+class FlyIoCheck(OpenMetricsBaseCheckV2, ConfigMixin):
+    DEFAULT_METRIC_LIMIT = 0
+    __NAMESPACE__ = 'fly_io'
+
+    def __init__(self, name, init_config, instances):
+        super().__init__(name, init_config, instances)
+        self.org_slug = self.instance.get('org_slug')
+        default_match_string = '{__name__=~".+"}'
+        self.match_string = self.instance.get('match_string', default_match_string)
```

Comment:
Why don't we use `self.config.match_string` since it uses the default value that we have declared in the yaml file?
If we use that approach we can remove the previous line.
Goal: Define the default value `'{__name__=~".+"}'` in just one place (`spec.yaml` file)
What do you think?